test/exe/version_out_test.sh

```diff
@@ -6,23 +6,29 @@ set -e -o pipefail
 export LC_ALL=C
 
 ENVOY_BIN="${TEST_SRCDIR}/envoy/source/exe/envoy-static"
+REGEX=".*version: ([0-9a-f]{40})\/([0-9]+\.[0-9]+\.[0-9]+)-([0-9]+)-g([0-9a-f]{10})(-dirty)?\/(Clean|Modified)\/(RELEASE|DEBUG)\/([a-zA-Z-]+)$"
 
-COMMIT=$(${ENVOY_BIN} --version | \
-  sed -n -E 's/.*version: ([0-9a-f]{40})\/([0-9]+\.[0-9]+\.[0-9]+)(-[a-zA-Z0-9\-_]+)?\/(Clean|Modified)\/(RELEASE|DEBUG)\/([a-zA-Z-]+)$/\1/p')
+COMMIT=$(${ENVOY_BIN} --version | sed -n -E "s/${REGEX}/\1/p")
+EXPECTED_COMMIT=$(cat "${TEST_SRCDIR}/envoy/bazel/raw_build_id.ldscript")
```

Comment:
It's a great idea to make the code more readable by naming it more explicitly. My only concern is that this patch will be merged to all future new releases, so I'd hope to keep the changes to be minimal and absolutely necessary to avoid potential merge conflicts.

How do you feel about keeping the old regex, make the necessary change from `/2/3` to `/2` and test the label format of `-g([0-9a-f]{10})(-dirty)` separately at the end of the file? Thus the merge conflicts will be obvious to fix.

Reply:
During testing, the old regex for the label in this test doesn't match the version string correctly, so it can't identify the commit or version. I think the multiple hyphens maybe throw it off. 

Note: This regex varies from the one used to make the build version [here](https://github.com/10gen/envoy-serverless/blob/develop/source/common/version/version.cc#L68), which uses `(-(.*))?` for the label capture group, considering everything after the first hyphen to be the label group.

I can revert the variable re-naming here to limit the diff here though

Reply:
I found the problem. Hyphen has to be the last in `[...]`: https://stackoverflow.com/a/4068725/1040243

Let's keep the hyphen fix in a separate commit and make a PR upstream later just for that. I don't think we have to add tests for the hyphen fix. An example of `echo ... | sed ...` would suffice in the upstream PR, I think.

Reply:
Made a PR to envoy [here](https://github.com/envoyproxy/envoy/pull/25124) for the test fix